src/aten/sycl/Indexing.h

```diff
@@ -1,17 +1,358 @@
 #pragma once
 
-#include <ATen/native/TensorIterator.h>
+#include <aten/sycl/BatchKernel.h>
+#include <comm/TensorInfo.h>
+
+using namespace at::xpu::detail;
+using namespace at::xpu;
 
 namespace at::native::xpu {
 
-void masked_fill_kernel(TensorIterator& iter, const Scalar& value);
+// Pretend that the scalar tensor is in fact a one-element vector.
+template <typename T, typename IndexType>
+TensorInfo<T, IndexType> tensorInfoIfScalar(TensorInfo<T, IndexType> ti) {
+  if (ti.dims == 0) {
+    ti.dims = 1;
+    ti.sizes[0] = 1;
+    ti.strides[0] = 1;
+  }
+  return ti;
+}
+
+template <class SrcInfo, class DstInfo, class IdxInfo, class FuncType>
+class IndexKernelConfig : public BatchKernelConfig {
+ public:
+  using ValType = typename SrcInfo::scalar_t;
+  using IdxType = typename IdxInfo::scalar_t;
+
+  IndexKernelConfig() = delete;
+  IndexKernelConfig(
+      SrcInfo& sinfo,
+      DstInfo& dinfo,
+      IdxInfo& iinfo,
+      ValType alpha,
+      int64_t index_num,
+      int64_t indexing_dimension_size,
+      bool indexing_dst,
+      bool problem_inner,
+      FuncType func,
+      int64_t batch,
+      int64_t problem,
+      int64_t stride,
+      int64_t problem_batch,
+      bool problem_along_x)
+      : BatchKernelConfig(
+            batch,
+            problem,
+            stride,
+            problem_batch,
+            problem_along_x,
+            Policy::pSegment,
+            syclMaxWorkItemsPerEU()),
+        sinfo_(sinfo),
+        dinfo_(dinfo),
+        iinfo_(iinfo),
+        alpha_(alpha),
+        index_num_(index_num),
+        indexing_dimension_size_(indexing_dimension_size),
+        indexing_dst_(indexing_dst),
+        problem_inner_(problem_inner),
+        func_(func) {}
+
+  template <class TarInfo>
```

Comment:
What's the meaning of `Tar`?

Reply:
Since it is a general kernel template shared among multiple indexing operators, some operator indexes along src tensor and some operators indexes along dst tensor. To give an abstract name, using `target` (`Tar`), which means the target tensor to index.

Reply:
Regarding these non-common abbreviations, I recommend keeping the full word for better code understanding. I would prefer to ensure the code self is the comments.